src/dragonball/src/dbs_virtio_devices/src/mmio/mmio_v2.rs

```diff
@@ -25,6 +27,24 @@ const DEVICE_STATUS_DRIVER: u32 = DEVICE_STATUS_ACKNOWLEDE | DEVICE_DRIVER;
 const DEVICE_STATUS_FEATURE_OK: u32 = DEVICE_STATUS_DRIVER | DEVICE_FEATURES_OK;
 const DEVICE_STATUS_DRIVER_OK: u32 = DEVICE_STATUS_FEATURE_OK | DEVICE_DRIVER_OK;
 
+/// mmio transport associated metrics.
+#[derive(Default, Serialize)]
+pub struct MMIODeviceMetrics {
+    /// Number of times when interacting with the space config of a mmio device failed.
+    pub cfg_fails: SharedIncMetric,
+    /// Number of times when handling read events on a mmio device.
+    pub read_count: SharedIncMetric,
+    /// Number of times when handling write events on a mmio device.
+    pub write_count: SharedIncMetric,
+    /// Number of times when handling read events on a mmio device failed.
+    pub read_fails: SharedIncMetric,
+    /// Number of times when handling write events on a mmio device failed.
+    pub write_fails: SharedIncMetric,
+    /// Number of times when activated on a mmio deivce.
+    pub activate_count: SharedIncMetric,
```

Comment:
The activate action only need to do once, we don't need to add this metric

Reply:
Same as other devices

Reply:
you can use an atomic variable to represent not activated, activated success or fail,

Reply:
Considering the possibility of multiple failed activations, I retained the `activate_fails`. And I used `activate_flag` to represent the state.
- If `activate_flag` is 1, it means the device is activated.
- If `activate_flag` is 0 and `activate_fails` is 0, it means the device is not activated.
- If `activate_flag` is 0 and `activate_fails` is not 0, it means the device has failed to activate.

Reply:
Great job!